src/main/scala/viper/gobra/frontend/info/implementation/resolution/Enclosing.scala

```diff
@@ -143,4 +144,60 @@ trait Enclosing { this: TypeInfoImpl =>
 
     aux(nil)
   }
+
+  // finds all used, modified and declared variables
+  private lazy val variableAnalysis: Vector[PStatement] => (Set[PIdnNode], Set[PIdnNode], Set[PIdnNode]) =
+    attr[Vector[PStatement], (Set[PIdnNode], Set[PIdnNode], Set[PIdnNode])] {stmts => 
+
+      def idNodeInAssignee(ass: PAssignee) =
+        allChildren(ass).find {
+          case _: PIdnNode => true 
+          case _ => false
+        }
```

Comment:
This looks wrong to me, e.g. `*m = 5`. the variable m is not modified.

Reply:
You are right. However, I would argue that this is just an over-approximation; in your example, m would just be reassigned the unchanged value of m that was used in the outline block. This results in one unnecessary assignment but is still technically correct.